packages/ui/src/bounty/modals/AnnounceWorkEntryModal/machine.ts

```diff
@@ -9,10 +10,16 @@ import {
 } from '@/common/model/machines'
 import { EmptyObject } from '@/common/types'
 
-interface TransactionContext {
+interface ContributionContext {
+  stakingAccount: Account
+}
+
+interface TransactionContext extends ContributionContext {
   transactionEvents?: EventRecord[]
 }
 
+type AnnounceWorkEntryContext = Partial<ContributionContext & TransactionContext>
```

Comment:
AFAICS one of these types is redundant. Why not:
```suggestion
interface ContributionContext {
  stakingAccount?: Account
}

interface TransactionContext extends ContributionContext {
  transactionEvents?: EventRecord[]
}
```
And later:
```ts
  | { value: AnnounceWorkEntryStates.contribute; context: Reqired<ContributionContext> }
```